src/audio_worklet.js

```diff
@@ -25,8 +25,8 @@ function createWasmAudioWorkletProcessor(audioParams) {
       assert(opts.callback)
       assert(opts.samplesPerChannel)
 #endif
-      this.callback = getWasmTableEntry(opts.callback);
-      this.userData = opts.userData;
+      this.callback = getWasmTableEntry({{{ toIndexType("opts.callback") }}});
+      this.userData = {{{ toIndexType("opts.userData") }}};
```

Comment:
This second line doesn't look quite right to me.

userData I assume is a pointer, which means it should be Number and not a bigint (it should be converted to a Number when it first arrived on the JS side).   

If it needs to be converted back to a pointer (i64) when `this.callback` then you should be using `to64` at the callback call side, and not here.

`toIndexType` should only be used for places where you are direclty accessing the table, or the memory bounds.   

Can you test with `-sMEMORY64=2` as well perhaps to catch these cases where there is difference between `to64` and `toIndexType`?

Reply:
Tested manually with `-sMEMORY64=2` (and `-sMEMORY64=1`) with most of the conversions now removed.